testing/api/test_qmri.py

```diff
@@ -20,7 +20,7 @@ def make_sct_image(data):
     data: scalar
     """
     affine = np.eye(4)
-    nii = nibabel.nifti1.Nifti1Image(np.array([data, data]), affine)
+    nii = nibabel.nifti1.Nifti1Image(np.array([data, data], dtype=np.uint32), affine)
```

Comment:
Re: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4408

> I encountered this exact error on https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4332, too. (See: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4332/commits/cb7157ff4c202dd87b3ce59efdd4826897bdd413) So, the errors will be fixed when #4332 is merged.
> 
> However, given how "general" this error message is, I am concerned that there will be instances of this issue elsewhere in our code that aren't caught by the test suite. So, I'd like to consider adding a fix to the `Image` constructor (either by passing `dtype=np.int64` when the input array is of type `np.int64` to preserve the old behavior, or by converting input arrays to `np.int32` before loading them to fix the potential incompatibilities hinted at by the code).

I realize now that both of these tests (`test_qmri.py`, `test_deepseg_sc.py`) call the `Nifti1Image` constructor directly, bypassing any fixes we might add to `Image`. So, in order to test a fix that gets added to the `Image` constructor, I will also need to rewrite these dummy test fixtures.

Reply:
Ah, wait! If I rewrite these tests to pass the problematic `np.int64` array directly to the `Image` constructor, then there are no failures, even without an explicit `np.int64` fix!

This is because, when we initialize an `Image` object from a numpy array, we never actually call `nib.Nifti1Image`. Instead, we simply store the numpy array and a separate `Nifti1Header` object.

We do call `nib.Nifti1Image` when _saving_ an `Image` object. But, we explicitly call it with the `header` argument, always. And, by specifying `header`, we bypass the check (as mentioned by the error message):

> To use this type, pass an explicit header or dtype argument to Nifti1Image()

So, as long as we're specifying the header for all uses of `Nifti1Image`, we can trust that there won't be any unexpected failures down the line.

(Though, we should probably still heed nibabel's compatibility warning anyway!)

Reply:
I think I'm going to mark this as resolved in the scope of this PR, but perhaps leave #4408 open to consider https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4408#issuecomment-2007396098.

Reply:
Thanks for the investigation!